eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/AbstractEventStorageEngine.java

```diff
@@ -314,8 +315,24 @@ public Builder eventSerializer(Serializer eventSerializer) {
          *
          * @param snapshotFilter a {@link Predicate} which decides whether to take a snapshot into account
          * @return the current Builder instance, for fluent interfacing
+         * @deprecated in favor of {@link #snapshotFilter(SnapshotFilter)}
          */
+        @Deprecated
         public Builder snapshotFilter(Predicate<? super DomainEventData<?>> snapshotFilter) {
+            return snapshotFilter(snapshotFilter::test);
```

Comment:
We had an `assertNonNull` before, should we keep it?

Reply:
Well, this method delegates to the `snapshotFilter(SnapshotFilter)` method, which does have the `assertNonNull`. So, we should be covered on the matter.